samples/bluetooth/direct_test_mode/src/transport/hci_uart_remote.c

```diff
@@ -0,0 +1,147 @@
+/*
+ * Copyright (c) 2023 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <zephyr/kernel.h>
+#include <zephyr/init.h>
+#include <zephyr/net/buf.h>
+#include <zephyr/logging/log.h>
+#include <nrf_rpc/nrf_rpc_ipc.h>
+#include <nrf_rpc_cbor.h>
+
+#include "hci_uart.h"
+#include "dtm_serialization.h"
+
+LOG_MODULE_REGISTER(serialize_layer);
+
+NRF_RPC_IPC_TRANSPORT(hci_group_tr, DEVICE_DT_GET(DT_NODELABEL(ipc0)), "nrf_rpc_ept");
+NRF_RPC_GROUP_DEFINE(hci_group, "hci_remote", &hci_group_tr, NULL, NULL, NULL);
+
+NET_BUF_POOL_DEFINE(tx_buf, CONFIG_DTM_HCI_QUEUE_COUNT, CONFIG_DTM_HCI_QUEUE_SIZE, 0, NULL);
+
+static hci_uart_read_cb callback;
+
+static void rsp_error_code_handle(const struct nrf_rpc_group *group, struct nrf_rpc_cbor_ctx *ctx,
+				  void *handler_data)
+{
+	int32_t val;
+
+	if (zcbor_int32_decode(ctx->zs, &val)) {
+		*(int *)handler_data = (int)val;
+	} else {
+		*(int *)handler_data = -NRF_EINVAL;
+	}
+}
+
+/* Incoming event from application core (uart) */
+static void dtm_hci_put_handler(const struct nrf_rpc_group *group, struct nrf_rpc_cbor_ctx *ctx,
+				void *handler_data)
+{
+	struct net_buf *buf;
+	uint8_t type;
+	struct zcbor_string bufzs;
+
+	LOG_DBG("Call from dtm_hci_put");
+
+	if (!zcbor_uint_decode(ctx->zs, &type, sizeof(type))) {
+		goto error_exit;
+	}
+
+	if (!zcbor_bstr_decode(ctx->zs, &bufzs)) {
+		goto error_exit;
+	}
+
+	buf = net_buf_alloc(&tx_buf, K_NO_WAIT);
+	net_buf_add_mem(buf, bufzs.value, bufzs.len);
+	buf->user_data[0] = type;
+
+	nrf_rpc_cbor_decoding_done(group, ctx);
+
+	callback(buf);
+	return;
+
+error_exit:
+	__ASSERT_NO_MSG(0);
```

Comment:
Please change it in all places
```suggestion
	__ASSERT_NO_MSG(false);
```